packages/platform/src/types.ts

```diff
@@ -24,14 +24,22 @@ export type Event<EventType> = {
   facts?: any
 }
 
-export type Output<OutputType, NotificationType, EventType> = {
-  type: OutputType
+// TODO(JJ): remove isDirty for ActionReducer
+export type OutputFactsUser<FactsType, NotificationType, EventType> = {
+  result: FactsType
+  isDirty?: boolean
   extras?: any
-  result: any
+  actions?: any[]
   notifications?: Notification<NotificationType>[]
   events?: Event<EventType>[]
+}
+
+export type OutputFacts<FactsType, NotificationType, EventType> = {
```

Comment:
I was wondering if it's better to shorten the names, e.g.: 
- OutputFacts -> OFacts
- InputFacts -> IFacts
- General Types -> TFacts
- etc.

Reply:
IFacts might be understood as an "interface". I would keep the naming in the extended form